framework/src/tinyalsa/tinyalsa.c

```diff
@@ -1320,54 +1412,48 @@ int pcm_wait(struct pcm *pcm, int timeout)
 			}
 		}
 		if (msg.msgId == AUDIO_MSG_DEQUEUE) {
+			/* If you find a dequeue message in the queue,
+			update the apb flags and curbyte and increment the count*/
 			apb = (struct ap_buffer_s *)msg.u.pPtr;
-			pcm->next_buf = apb;
-			pcm->next_size = apb->nbytes;
-			pcm->buf_ptr--;
+			apb->flags &= ~AUDIO_APB_OUTPUT_ENQUEUED;
+			apb->curbyte = 0;
```

Comment:
if (msg.msgId == AUDIO_MSG_DEQUEUE) {
  .....
  return 1;
}
return oops;

Reply:
Ok. I changed it.